### DISCUSSION TITLE
allow implicit target type deduction

### DISCUSSION THREAD

**Discussion by kasperk81** [dotnet/roslyn#75217] [Ideas] [answered]
`foreach (var n in [1,2,3])` rightfully throws `error CS9176: There is no target type for the collection expression.` and expects `(int[])` cast.

compiler should at least be able to deduce int[] implicitly when type is specified `foreach (int n in [1,2,3])`?

**Comment by CyrusNajmabadi** [comment, marked as answer]
This is not a compiler issue. This is a request to expand what the language supports. Those requests are tracked at dotnet/csharplang.  Thanks!
